src/views/Article/Index.jsx

```diff
@@ -51,21 +60,45 @@ const Article = (props) => {
     }
   } = props;
   const shareUrl = `https://vidar-ah-frontend-staging.herokuapp.com/articles/${slug}`;
+
+  const handleScroll = () => {
+    const { innerHeight } = window;
+    const { documentElement: { scrollTop } } = document;
+    const bodyElement = document.getElementById('article-container');
+    const clientDocument = bodyElement.getBoundingClientRect();
+    const heightIsHtml = clientDocument.height;
+    const scrollMax = Math.ceil(heightIsHtml - innerHeight);
+    setProgress((scrollTop / scrollMax) * 100);
+  };
+
   useEffect(() => {
     getArticle(slug);
     getArticles();
+    window.addEventListener('scroll', handleScroll);
```

Comment:
@Dubby20 what is this doing?

Reply:
It updates the progress bar when scrolling the page